File: app/src/main/java/com/lazycouple/restapiclient/db/model/Parameter.java
package com.lazycouple.restapiclient.db.model;

import io.realm.RealmObject;

/**
 * Created by noco on 2016-10-13.
 */
public class Parameter extends RealmObject {
    private String id;
    private String key;
    private String value;

    public Parameter() {
    }

    public Parameter(String key, String value) {
        this.key = key;
        this.value = value;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "Parameter{" +
                "id='" + id + '\'' +
                ", key='" + key + '\'' +
                ", value='" + value + '\'' +
                '}';
    }

    public static final class Builder {
        private String key;
        private String value;


        public Builder key(String key) {
            this.key = key;
            return this;
        }

        public Builder value(String value) {
            this.value = value;
            return this;
        }

        public Parameter build() {
            Parameter parameter = new Parameter();
            parameter.setKey(key);
            parameter.setValue(value);
            return parameter;
        }
    }
}

File: app/src/main/java/com/lazycouple/restapiclient/db/model/Parameter.kt
package com.lazycouple.restapiclient.db.model

import io.realm.RealmObject

/**
 * Created by noco on 2016-10-13.
 */
open class Parameter : RealmObject {
    var id: String? = null
    var key: String? = null
    var value: String? = null

    constructor() {}

    constructor(key: String, value: String) {
        this.key = key
        this.value = value
    }

    override fun toString(): String {
        return "Parameter{" +
                "id='" + id + '\'' +
                ", key='" + key + '\'' +
                ", value='" + value + '\'' +
                '}'
    }

    class Builder {
        private var key: String? = null
        private var value: String? = null


        fun key(key: String): Builder {
            this.key = key
            return this
        }

        fun value(value: String): Builder {
            this.value = value
            return this
        }

        fun build(): Parameter {
            val parameter = Parameter()
            parameter.key = key
            parameter.value = value
            return parameter
        }
    }
}
